pgrx/src/datum/datetime_support/ctor.rs

```diff
@@ -85,45 +85,3 @@ pub fn to_timestamp(epoch_seconds: f64) -> TimestampWithTimeZone {
         direct_function_call(pg_sys::float8_timestamptz, &[epoch_seconds.into_datum()]).unwrap()
     }
 }
-
-/// “bins” the input timestamp into the specified interval (the stride) aligned with a specified origin.
-///
-/// `source` is a value expression of type [`Timestamp`].
-/// `stride` is a value expression of type [`Interval`].
-///
-/// The return value is likewise of type [`Timestamp`], and it marks the beginning of the bin into
-/// which the source is placed.
-///
-/// # Notes
-///
-/// Only available on Postgres v14 and greater.
-///
-/// In the case of full units (1 minute, 1 hour, etc.), it gives the same result as the analogous
-/// `date_trunc()` function, but the difference is that [`date_bin()`] can truncate to an arbitrary
-/// interval.
-///
-/// The stride interval must be greater than zero and cannot contain units of month or larger.
-///
-/// # Examples
-///
-/// ```sql
-/// SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01');
-/// Result: 2020-02-11 15:30:00
-///
-/// SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01 00:02:30');
-/// Result: 2020-02-11 15:32:30
-/// ```
-#[cfg(any(features = "pg14", features = "pg15"))]
-pub fn date_bin(
-    stride: crate::datum::interval::Interval,
-    source: Timestamp,
-    origin: Timestamp,
-) -> Timestamp {
-    unsafe {
-        direct_function_call(
-            pg_sys::date_bin,
```

Comment:
haha.  This should have been `timestamp_bin` and/or `timestamptz_bin`

per Discord chat, we can instead add a `bin(stride, origin)` function to those two types instead.